firedrake/matrix_free/operators.py

```diff
@@ -251,31 +251,36 @@ def multTranspose(self, mat, Y, X):
         * = can be any number
 
         """
-        # accumulate values in self._xbc for convenience
-        self._xbc.dat.zero()
         with self._y.dat.vec_wo as v:
             Y.copy(v)
 
-        # Apply actionTs in sorted order
-        for aT, obj in zip(self._assemble_actionT, self.objs_actionT):
-            # zero columns associated with DirichletBCs/EquationBCs
-            for obc in obj.bcs:
-                obc.zero(self._y)
-            aT()
-            self._xbc += self._x
+        if len(self.bcs) > 0:
+            # Accumulate values in self._x
+            self._x.dat.zero()
+            # Apply actionTs in sorted order
+            for aT, obj in zip(self._assemble_actionT, self.objs_actionT):
+                # zero columns associated with DirichletBCs/EquationBCs
+                for obc in obj.bcs:
+                    obc.zero(self._y)
+                aT()
+                self._x += self._xbc
+        else:
+            # No DirichletBC/EquationBC
+            # Save to self._x directly
+            self._assemble_actionT[0]()
```

Comment:
Loop over the list here and call all of them? Just in case there's ever more than one.